### PR TITLE
move GNPSExport to lib

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

Moves GNPSExport functionality to library. Some smaller fixes.

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
rebuild jenkins

File: src/openms/include/OpenMS/FORMAT/GNPSMGFFile.h [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+{
+  class OPENMS_DLLAPI GNPSMGFFile : 
+    public DefaultParamHandler,
+    public ProgressLogger
+  {
+    public:
+      // default c'tor
+      GNPSMGFFile();
+
+      // see GNPSExport tool documentation
```
Comment (hroest) [inline src/openms/include/OpenMS/FORMAT/GNPSMGFFile.h:51]: can you add some minimal documentation, at least what the parameters are and what to expect? How would these mzml and consensus files be generated? Is the output a spectral library?

**Comment by jpfeuffer** [comment]
```
Error compiling Cython file:
------------------------------------------------------------
...
        #  DefaultParamHandler        

        GNPSMGFFile() nogil except +
        GNPSMGFFile(GNPSMGFFile &) nogil except +

        void run(String & consensus_file_path, StringList & mzml_file_paths, String & out) nogil except + # wrap-doc:Export consensus file from default workflow to GNPS MGF format
                                              ^
------------------------------------------------------------
```

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
tests passed

**Pull request merged**